### PR TITLE
Sprint poker template UI (minus scale UI)

### PR DESCRIPTION
**Pull request by tianrunhe**
Issue: #4057 

This PR implements UI for sprint poker template minus the scale UI part.

Tests:

- [ ] Add a new poker template
- [ ] Clone a poker template
- [ ] Rename a poker template
- [ ] Delete a poker template
- [ ] Change the scope of a poker template
- [ ] Add a new dimension
- [ ] Rename a dimension
- [ ] Delete a dimension
- [ ] Drag and move a dimension
- [ ] Select a scale for a dimension

### CODE REVIEW

**Comment by tianrunhe** [comment]
@ackernaut Hey Terry, would love to have your first pass review before I send it to Matt. Thanks in advance!

**Comment by ackernaut** [comment]
@tianrunhe will do, thanks for pinging!

**Comment by ackernaut** [comment]
### Test no. 1
- [x] Add a new poker template
- [ ] Clone a poker template
- [ ] Rename a poker template
- [ ] Delete a poker template
- [ ] Change the scope of a poker template
- [ ] Add a new dimension
- [ ] Rename a dimension
- [ ] Delete a dimension
- [ ] Drag and move a dimension
- [ ] Select a scale for a dimension

**Review by ackernaut — CHANGES_REQUESTED** [review]
Nice work, Bruce!! Thanks for the complete checklist. The UI is working well. For some reason, I’m seeing duplicate scale items in the public template (see screenshot). Also, I pushed a small change to how those hanging components are positioned on template items as I noticed small nuances in how we were laying out icons etc. Once you figure out the duplicate scale items, I’d say it’s ready for code review.
Screen Shot 2020-11-05 at 13 06 28

**Comment by tianrunhe** [comment]
@ackernaut Thanks for the review! A quick fix was just pushed, please verify :smiley:

**Comment by ackernaut** [comment]
@tianrunhe that change worked for me! I noticed that when I open the scale menu some selected options are already highlighted and others aren’t. See here:

**Review by ackernaut — APPROVED** [review]

**Comment by tianrunhe** [comment]
> @tianrunhe that change worked for me! I noticed that when I open the scale menu some selected options are already highlighted and others aren’t. See here:
> 
> 

That's an interesting find! I wonder if that's an existing implicit bug in our `Menu` implementation. The symptom is that you got to have a `DropdownMenuLabel` component inside the `Menu` component, in order for the indices work. Otherwise, like in this case, the active indices will be correct except for the very first one. I'll talk with Matt on this one.

File: packages/client/modules/meeting/components/AddNewPokerTemplate.tsx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  useEffect(() => {
+    return () => {
+      window.clearTimeout(errorTimerId.current)
+    }
+  }, [])
+  const addNewTemplate = () => {
+    if (submitting) return
+    if (pokerTemplates.length >= Threshold.MAX_RETRO_TEAM_TEMPLATES) {
+      onError(new Error('You may only have 20 templates per team. Please remove one first.'))
+      errorTimerId.current = window.setTimeout(() => {
```
Comment (mattkrick) [inline packages/client/modules/meeting/components/AddNewPokerTemplate.tsx:56]: this is a really nice touch!
what do you think about refactoring this logic into `useMutationProps`?
API could be `onError(error: Error, removeAfter: number)`

File: packages/client/modules/meeting/components/AddPokerTemplateDimension.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  display: 'block',
+  margin: '0 16px 0 16px'
+})
+
+interface Props extends WithAtmosphereProps, WithMutationProps {
+  dimensions: AddPokerTemplateDimension_dimensions
+  templateId: string
+}
+
+class AddPokerTemplateDimension extends Component<Props> {
```
Comment (mattkrick) [inline packages/client/modules/meeting/components/AddPokerTemplateDimension.tsx:38]: nit: can you write this as a functional component?
we deprecated class-based components back in 2018 & have been slowly trying to get them out of the app. Once they're gone we can get rid of a lot of class-only helpers like `withAtmosphere` so we can keep our codebase as small as possible

File: packages/client/modules/meeting/components/EditableTemplateDimension.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+interface Props extends WithAtmosphereProps, WithMutationProps {
+  isOwner: boolean
+  isEditingDescription: boolean
+  isHover: boolean
+  dimensionName: string
+  dimensionId: string
+  dimensions: EditableTemplateDimension_dimensions
+}
+
+class EditableTemplateDimension extends Component<Props> {
```
Comment (mattkrick) [inline packages/client/modules/meeting/components/EditableTemplateDimension.tsx:31]: functional component, if possible

**Review by mattkrick — CHANGES_REQUESTED** [review]

File: packages/client/modules/meeting/components/AddPokerTemplateDimension.tsx [hunk 4]
```diff
@@ -63,16 +63,14 @@ class AddPokerTemplateDimension extends Component<Props> {
...
+  if (dimensions.length >= Threshold.MAX_REFLECTION_PROMPTS) return null
+  return (
+    <AddDimensionLink palette='blue' onClick={addDimension} waiting={submitting}>
+      <AddDimensionLinkPlus>add</AddDimensionLinkPlus>
+      <div>Add another dimension</div>
+    </AddDimensionLink>
+  )
 }
 
 export default createFragmentContainer(withMutationProps(withAtmosphere(AddPokerTemplateDimension)), {
```
Comment (mattkrick) [inline packages/client/modules/meeting/components/AddPokerTemplateDimension.tsx:76]: ah, for the refactor can you repace the HOCs with hooks? 
e.g. get rid of `withAtmosphere` and add `useAtmosphere`

**Comment by mattkrick** [comment]
CR complete! going through the UI:
- [ ] for disabled scale dropdowns, at a minimum we need to have `cursor: disabled`, i kept clicking it wondering why it wasn't dropping down. additionally, we could remove the down arrow icon, or make the text a little more gray. @ackernaut 
- [ ] for retros, the public templates that don't have illustrations show nothing (I think this is a bug). For poker, the ones that don't have illustrations show the default picture. they should act the same

**Comment by mattkrick** [comment]
🎉  nice work!

**Pull request merged**